lib/index.js

```diff
@@ -480,6 +480,13 @@ InputMask.prototype.getValue = function getValue() {
   return this.value.join('')
 }
 
+InputMask.prototype.hasValueChanged = function hasValueChanged(nextValue) {
+  if (!nextValue) {
+    nextValue = ''
+  }
+  return this.value.join('') !== this.pattern.formatValue(nextValue.split('')).join('')
```

Comment:
Few months since I looked at this, but I think it's because InputMask stores its internal representation of the field data as an array.  Don't recall exactly why, but all related to how it processes the mask characters I believe.